ResturantReservation.Db/Data/RestaurantReservationDbContext.cs

```diff
@@ -0,0 +1,102 @@
+            );
+
+            modelBuilder.Entity<Reservation>().HasData(
+                new Reservation { ReservationId = 1, ResturantId = 1, CustomerId = 1, PartySize = 3, TableId = 1, ReservationDate = new DateTime(2024, 8, 12) },
+                new Reservation { ReservationId = 2, ResturantId = 2, CustomerId = 2, PartySize = 2, TableId = 2, ReservationDate = new DateTime(2024, 8, 13) },
+                new Reservation { ReservationId = 3, ResturantId = 3, CustomerId = 3, PartySize = 4, TableId = 3, ReservationDate = new DateTime(2024, 8, 14) },
+                new Reservation { ReservationId = 4, ResturantId = 4, CustomerId = 4, PartySize = 5, TableId = 4, ReservationDate = new DateTime(2024, 8, 15) },
+                new Reservation { ReservationId = 5, ResturantId = 5, CustomerId = 5, PartySize = 6, TableId = 5, ReservationDate = new DateTime(2024, 8, 16) }
+            );
+
+            modelBuilder.Entity<Order>().HasData(
+                new Order { OrderId = 1, EmployeeId = 1, ReservationId = 1, TotalAmount = 20, OrderDate = new DateTime(2024, 7, 13) },
+                new Order { OrderId = 2, EmployeeId = 2, ReservationId = 2, TotalAmount = 30, OrderDate = new DateTime(2024, 7, 14) },
+                new Order { OrderId = 3, EmployeeId = 3, ReservationId = 3, TotalAmount = 40, OrderDate = new DateTime(2024, 7, 15) },
+                new Order { OrderId = 4, EmployeeId = 4, ReservationId = 4, TotalAmount = 50, OrderDate = new DateTime(2024, 7, 16) },
+                new Order { OrderId = 5, EmployeeId = 5, ReservationId = 5, TotalAmount = 60, OrderDate = new DateTime(2024, 7, 17) }
+            );
+
+            modelBuilder.Entity<MenuItem>().HasData(
+                new MenuItem { MenuItemId = 1, ResturantId = 1, Name = "Potato", Description = "Crunchy", Price = 10 },
+                new MenuItem { MenuItemId = 2, ResturantId = 2, Name = "Pasta", Description = "Cheesy", Price = 15 },
+                new MenuItem { MenuItemId = 3, ResturantId = 3, Name = "Pizza", Description = "Delicious", Price = 20 },
+                new MenuItem { MenuItemId = 4, ResturantId = 4, Name = "Burger", Description = "Juicy", Price = 25 },
+                new MenuItem { MenuItemId = 5, ResturantId = 5, Name = "Sushi", Description = "Fresh", Price = 30 }
+            );
+
+            modelBuilder.Entity<OrderItem>().HasData(
+                new OrderItem { OrderItemId = 1, OrderId = 1, MenuItemId = 1, Quantity = 2 },
+                new OrderItem { OrderItemId = 2, OrderId = 2, MenuItemId = 2, Quantity = 3 },
+                new OrderItem { OrderItemId = 3, OrderId = 3, MenuItemId = 3, Quantity = 4 },
+                new OrderItem { OrderItemId = 4, OrderId = 4, MenuItemId = 4, Quantity = 5 },
+                new OrderItem { OrderItemId = 5, OrderId = 5, MenuItemId = 5, Quantity = 6 }
+            );
+
+            modelBuilder.Entity<ReservationDetail>().ToView("ReservationDetails").HasNoKey();
```

Comment:
view result should not be tracked by EF Core because they're not actual entities. does the `HasNoKey` automatically exclude the results from the tracker or we still need to use `AsNoTracking` when querying it?